infra/main.bicep

```diff
@@ -138,6 +141,31 @@ param azureOpenAIEmbeddingModel string = 'text-embedding-ada-002'
 @description('Azure OpenAI Embedding Model Name')
 param azureOpenAIEmbeddingModelName string = 'text-embedding-ada-002'
 
+@description('Azure OpenAI Embedding Model Capacity - See here for more info  https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/quota')
+param azureOpenAIEmbeddingModelCapacity int = 30
+
+@description('Name of Computer Vision Resource (if using GPT-4 and Vision Preview)')
+param computerVisionName string = 'computer-vision-${resourceToken}'
+
+@description('Name of Computer Vision Resource SKU (if using GPT-4 and Vision Preview)')
+@allowed([
+  'F0'
+  'S1'
+])
+param computerVisionSkuName string = 'S1'
+
+@description('Location of Computer Vision Resource (if using GPT-4 and Vision Preview)')
+@allowed([// List taken from https://learn.microsoft.com/en-us/azure/ai-services/computer-vision/how-to/image-retrieval?tabs=python#prerequisites
+  'eastus'
+  'westus'
+  'koreacentral'
+  'francecentral'
+  'northeurope'
+  'westeurope'
+  'southeastasia'
+])
+param computerVisionLocation string = location
```

Comment:
If someone has previously deployed the application to a region not in this list, the next time they deploy they'll get an error, even if they're not using vision. Is there any way we can avoid this?

Reply:
Hmm, very very good catch! I guess one option is to not set a default, which would mean the user would always have to supply this. But that's not a great user exeperience for when they're not even going to be use Computer Vision. 

Maybe we should just skip validation of this for now? I'm not a huge fan of comments hidden in "code", but maybe that could be a best effort for now?

Reply:
Alternatively, I could create a new `computervision.bicep`, that calls `cognitiveservices.bicep` and validate the param in there. That way it'll only be called if it's enabled. 

I _think_ that will work.

WDYT?

Reply:
I get this error when I try and do that:

This symbol cannot be referenced here. Only other parameters can be referenced in parameter default values.bicep(BCP072)

Reply:
Ah but then we get this error:
Expected a value of type "'eastus' | 'francecentral' | 'koreacentral' | 'northeurope' | 'southeastasia' | 'westeurope' | 'westus'" but the provided value is of type "'' | string".bicep(BCP033)

Reply:
Add `''` to the allowed values

Reply:
> Add '' to the allowed values

I'm not sure I like that 😅  Lemme try the approach of moving the validation down

Reply:
Yea that doesn't work, lets go with adding `''` to the allowed values